### PR TITLE
Add linting including a gh action

### PR DESCRIPTION
**Pull request by PicoCentauri**
----
:books: Documentation preview :books:: https://software-cookbook--24.org.readthedocs.build/en/24/

### CODE REVIEW

**Comment by agoscinski** [comment]
Need to check if the conversion .ipynb -> .py sphinx example using the script is in agreement with sphinx-lint rules. Don't want to add friction for people contributing.

**Comment by PicoCentauri** [comment]
I took a look at it looks fine. There is already a check for the line width which the hardest part. Also we can say that people can use `tox -e format` before they create a PR.

**Comment by PicoCentauri** [comment]
@agoscinski I checked the converter and with a few adjustments it works super well. `pypandoc` automatically follows a linewidth of 88 chars. I also added some information how to format a new file automatically.

File: CONTRIBUTING.rst [hunk 1]
```diff
@@ -39,12 +41,19 @@ sphinx-gallery example. To convert your Jupyter notebook you can just use the
 Building the cookbook locally
 -----------------------------
 
-When you add a new example, you can build the doc and check if your code runs with
+When you add a new example, you can run the linter and build the doc to check if your
```
Comment (agoscinski) [inline CONTRIBUTING.rst]:
```suggestion
When you add a new example, you can run the linter (code format checker) and build the doc to check if your
```

File: CONTRIBUTING.rst [hunk 2]
```diff
@@ -39,12 +41,19 @@ sphinx-gallery example. To convert your Jupyter notebook you can just use the
...
 .. code-block:: bash
 
     tox
 
+If there are formatting errors appearing you can format your file automatically with
+
+.. code-block:: bash
+
+    tox -e format
+
```
Comment (agoscinski) [inline CONTRIBUTING.rst:56]:
```suggestion
That should fix most of the formatting issues automatically. If there are still formatting issues remaining, then the reviewer of your pull request can fix them.
```

**Review by agoscinski — COMMENTED** [review]
Thanks for the PR. Looks good. Just small comments

---
This PR added formatting and linting and a corresponding CI to it. How to use the formatting has been added to the contribution guidelines. There is noise in the PR in a lot of files, because the formater is applied on all files. The essential changes are in the files:
- CONTRIBUTING.rst
- pyproject.toml
- tox.ini
- .github/workflows/lint.yml

**Review by agoscinski — APPROVED** [review]
LGTM, just wait till CI passes. Let the linting pain begin.

**Pull request merged**